src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -2164,6 +2167,12 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 _terminal->Write(hstr);
             }
 
+            if (!_pendingResponses.empty())
+            {
+                _sendInputToConnection(_pendingResponses);
+                _pendingResponses.clear();
+            }

```

Comment:
This fixes the bug. It also fixes 2nd bug: We'd previously use the old `_sendInputToConnection` for responses but this would check whether we're read-only. We shouldn't do that for VT responses.

Reply:
oh this is clever. it replies with it immediately after the write that caused a reply?

Reply:
Yeah exactly. This is much better than before, because it's now processing an entire VT string atomically without unlocking in between. Not sure why I haven't done this from the get-go.